airflow/dag_context_manager/__init__.py

```diff
@@ -0,0 +1,23 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""DAG context manager used for convenient task creation."""
+# Used by DAG context_managers
+from typing import Optional
+
+from airflow.models.dag import DAG
+
+CONTEXT_MANAGER_DAG = None  # type: Optional[DAG]
```

Comment:
Rather than it's own file, might this make more sense in `airflow/models/dag.py` - it's very closely tied to `class DAG`

Reply:
Or even a class property on `class Dag` itself.

Reply:
Yeah. It might be in DAG indeed. I looked through the usages and it seems there is no risk of adding new cyclic imports if we make it part of DAG without adding risk of cyclic imports. This is classic step here - first you separate out something that is closely coupled with something else (settings in this case) just to find out that you can couple it with something else.

Reply:
Moved it as class instances/class methods of DAG. I am not big fan of class properties but I think it is mych nicer now anyway.